src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -30,4 +30,8 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
       case _ => Failure(new UnsupportedOperationException("Can only TSV serialize an Array[Primitive]"))
     }
   }
+
+  override def collect[T <: WdlValue](filterFn: PartialFunction[WdlValue, T]): Seq[T] = {
+    value flatMap { _.collect(filterFn) }
+  }
```

Comment:
Good catch. This isn't the first time something like this has come up. I'm not dead set against being sneaky like this but we should keep them to a dull roar and have a pretty good justification for it.